### PR TITLE
🌱 clusterctl: add in-cluster ipam provider

### PR DESCRIPTION
**Pull request by schrej**
**What this PR does / why we need it**:
Adds the in-cluster ipam provider to clusterctl.

Draft until we release version 0.1.0 of the provider.

### CODE REVIEW

File: cmd/clusterctl/client/config/providers_client.go [hunk 1]
```diff
@@ -315,6 +320,13 @@ func (p *providersClient) defaults() []Provider {
 			providerType: clusterctlv1.ControlPlaneProviderType,
 		},
 
+		// IPAM providers
```
Comment (sbueringer) [inline cmd/clusterctl/client/config/providers_client.go:323]: Let's also add the provider to the glossary etc.

For reference: https://github.com/kubernetes-sigs/cluster-api/pull/9012 (should be ~ the same)

**Comment by schrej** [comment]
> @schrej @rikatz Any plans of cutting a v0.1.0 release and adding the provider to clusterctl? :)

Yes. We still need to finish setting up some official image builds. I think we were waiting for a review, I'll take another look.

File: docs/book/src/reference/glossary.md [hunk 2]
```diff
@@ -108,6 +108,9 @@ Cluster API Provider VMware Cloud Director
 ### CAPZ
 Cluster API Provider Azure
 
+### CAIPAMIC
```
Comment (schrej) [inline docs/book/src/reference/glossary.md:111]: Not sure about `CAIPAM` as a prefix for ipam providers, but probably the best option next to `CAIP`

Comment (sbueringer) [reply docs/book/src/reference/glossary.md:111]: Sounds reasonable to me

**Comment by schrej** [comment]
That failed GH action seems to be a flake, that link is working fine for me.

**Comment by sbueringer** [comment]
Yeah saw that Azure link failing at least 10-20x times in the last month

**Comment by guettli** [comment]
> Yeah saw that Azure link failing at least 10-20x times in the last month

@sbueringer is someone working on that, so that we have stable tests?

**Comment by fabriziopandini** [comment]
/lgtm
/approve

🥳

**Comment by sbueringer** [comment]
> > Yeah saw that Azure link failing at least 10-20x times in the last month
> 
> @sbueringer is someone working on that, so that we have stable tests?

Not that I'm aware of

**Pull request merged**